pkg/util/http_client.go

```diff
@@ -0,0 +1,12 @@
+package util
+
+import (
+	"net/http"
+
+	otelhttp "go.opentelemetry.io/contrib/instrumentation/net/http"
+)
+
+// NewHTTPClient constructs an HTTP client with instrumentation.
+func NewHTTPClient() *http.Client {
```

Comment:
What are your thoughts on using this occasion to move the HTTPClient interface we have in pkg/blobstore to this file as well?

```go
// HTTPClient is an interface around Go's standard HTTP client type. It          
// has been added to aid unit testing.                                           
type HTTPClient interface {
        Do(req *http.Request) (*http.Response, error)
}
```

The convention so far is also to not use NewFoo() functions in case singleton instances suffice. We could provide a default HTTP client object using something like this:

```go
var DefaultHTTPClient HTTPClient = &http.Client{
        Transport: otelhttp.NewTransport(http.DefaultTransport),
}
```

Or are you dealing with certain pieces of code that strongly depend on *http.Client instead of using an interface type? If so, we could declare it without the `HTTPClient` in between and add a compile-time type assertion next to it.

Reply:
This turns out to work, done.